apps/deploy-web/src/context/TemplatesProvider/TemplatesProviderContext.tsx

```diff
@@ -1,28 +1,22 @@
 "use client";
 import React from "react";
 
-import { useTemplates as useTemplatesQuery } from "@src/queries/useTemplateQuery";
-import { ApiTemplate } from "@src/types";
+import { EnhancedTemplateCategory, TemplateOutputSummaryWithCategory, useTemplates as useTemplatesQuery } from "@src/queries/useTemplateQuery";
 
 type ContextType = {
   isLoading: boolean;
-  categories: Array<{ title: string; templates: Array<ApiTemplate> }>;
-  templates: Array<ApiTemplate>;
-  getTemplateById: (id: string) => ApiTemplate;
+  categories: EnhancedTemplateCategory[];
+  templates: TemplateOutputSummaryWithCategory[];
 };
 
 const TemplatesProviderContext = React.createContext<ContextType>({} as ContextType);
 
 export const TemplatesProvider = ({ children }) => {
   const { data, isFetching: isLoading } = useTemplatesQuery();
```

Comment:
Perhaps we could remove this context provider entirely and only call the query where we need to load the templates. That way it won't add more http requests globally when we might not need it.

Reply:
it's hard to judge from my standpoint since I don't have a full picture of the app. I'd merge this first, so we get performance benefits and only then do more refactoring if needed.

I saw that there are 3 http requests which are calling the same endpoint on page load. This obviously not good and needs to be optimized with caching and probably [async lock](https://developer.mozilla.org/en-US/docs/Web/API/Web_Locks_API) or globally on http service layer with traffic warden 

<img width="587" alt="Screenshot 2025-01-15 at 11 17 12" src="https://github.com/user-attachments/assets/868059cb-67c6-4e8a-92a7-3339af82f0cc" />

Reply:
I don't mind if you do it in another branch, however I think we should address this now rather than later as minor refactors like this tend to never get done.

Reply:
https://github.com/akash-network/console/issues/644